File: Pemrograman Berorientasi Objek/Materi/Pertemuan 6/Template/Titik.cpp
#include "Titik.h"

template <class T>
Titik<T>::Titik(){
	/*konstruktor*/
	Titik<T>::x = 0;
	Titik<T>::y = 0;
}

template <class T>
Titik<T>::Titik(int xp,int yp){
	/*konstruktor*/
	Titik<T>::x = xp;
	Titik<T>::y = yp;
}

template <class T>
int Titik<T>::getX(){
	/*mengembalikan nilai x*/
	return Titik<T>::x;
}

template <class T>
void Titik<T>::setX(int xp){
	/*mengeset nilai x*/
	Titik<T>::x = xp;
}

template <class T>
int Titik<T>::getY(){
	/*mengembalikan nilai y*/
	return Titik<T>::y;
}

template <class T>
void Titik<T>::setY(int yp){
	/*mengeset nilai x*/
	Titik<T>::y = yp;
}

template <class T>
Titik<T>::~Titik(){
}

File: Pemrograman Berorientasi Objek/Materi/Pertemuan 1/Konstruktor & Destructor/Titik.php
<?php

	class Titik{
		/*kelas yang digunakan untuk mengimplementasikan sebuah tipe titik*/
		
		private $x; /*koordinat x*/
		private $y; /*koordinat y*/
		
		public function __construct(){
			/*konstruktor*/
			$this->x = 0;
			$this->y = 0;
		}
		
		public function setX($xp){
			/*mengeset nilai koordinat x*/
			$this->x = $xp;
		}
		
		public function getX(){
			/*mengembalikan nilai koordinat x*/
			return $this->x;
		}
		
		public function setY($yp){
			/*mengeset nilai koordinat y*/
			$this->y = $yp;
		}
		
		public function getY(){
			/*mengembalikan nilai koordinat x*/
			return $this->y;
		}
		
		function __destruct(){
			/*destruktor*/
		}
		
	}


?>